lib/grape/middleware/base.rb

```diff
@@ -14,9 +14,9 @@ class Base
 
       # @param [Rack Application] app The standard argument for a Rack middleware.
       # @param [Hash] options A hash of options, simply stored for use by subclasses.
-      def initialize(app, **options)
+      def initialize(app, options = {})
```

Comment:
I think the backwards-compatible workaround solution would be:

```ruby
def initialize(app, options = nil)
  @app = app
  @options = default_options.dup
  @options.update(options) if options.is_a?(Hash)
  @app_response = nil
end
```

This retains the optional argument character of `options` while resolving the deprecation warning.

Reply:
And do we need it to be backwards compatible here?

Reply:
Defaulting the `options` parameter to an empty hash is backwards compatible. When declaring the parameter with `**options`, the `options` variable inside the method will default to an empty hash. 

See this example, run with Ruby 2.7.

```
def example_method(positional, **options)
  puts "  positional: #{positional}"
  puts "  options class: #{options.class}"
  puts "  options: #{options}\n"
end
```

```
Given positional argument only:
example_method("arg1")
  positional: arg1
  options class: Hash
  options: {}

Given positional argument followed by hash argument:
example_method("arg1", a_key: "a value")
  positional: arg1
  options class: Hash
  options: {:a_key=>"a value"}
```

Like the existing `initialize` method in `Grape::Middleware::Base`, the fact that `example_method` has `options` with the `**` in front declares it a keyword arguments, not as a (positional) hash argument. However the logic inside `Grape::Middleware::Base#initialize` treats `options` as a hash - it just merges it with `default_options`.

When passing a hash as a positional argument, Ruby 2.7 prints the deprecation warning because it's treating the given hash as keyword arguments in accordance with the method signature:

```
Given positional argument followed by positional hash:
example_method("arg1", { a_key: "a value" })
test.rb:15: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
test.rb:1: warning: The called method `example_method' is defined here
  positional: arg1
  options class: Hash
  options: {:a_key=>"a value"}
```

Changing the signature to `def example_method(positional, options = {})` makes the deprecation warning go away, but otherwise matches the same behavior for all of these examples. This is what I've done in this PR.